src/helpers.ts

```diff
@@ -64,11 +65,63 @@ export async function netrcPath(): Promise<string> {
         `If you're signed up for FlakeHub Cache, make sure that your Actions config has a \`permissions\` block with \`id-token\` set to "write" and \`contents\` set to "read"`,
       );
       actionsCore.debug(`Error while logging into FlakeHub: ${e}`);
+
+      await annotateWorkflowJobMissingPermissions();
     }
     return destinedNetrcPath;
   }
 }
 
+async function annotateWorkflowJobMissingPermissions(): Promise<void> {
+  try {
+    const workflowFileName = process.env["GITHUB_WORKFLOW"];
+    if (!workflowFileName) {
+      return;
+    }
+
+    const jobName = process.env["GITHUB_JOB"];
+    if (!jobName) {
+      return;
+    }
+
+    let matchedLines = 0;
+    let lineNumber = undefined;
+    const lines = (
+      await fs.readFile(workflowFileName, { encoding: "utf8" })
+    ).split("\n");
+    for (const lineIdx of lines.keys()) {
+      if (lines[lineIdx].includes(`${jobName}:`)) {
+        matchedLines += 1;
+        lineNumber = lineIdx + 1;
+      }
+    }
+
+    if (matchedLines !== 1) {
+      lineNumber = undefined;
+    }
```

Comment:
...if we matched multiple lines, we can't be too specific about what line we should be commenting on.